src/makeInfinityAdder.js

```diff
@@ -4,7 +4,29 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let i = 0;
+
+  const adder = (valueFirst) => {
+    if (i === 0 && !valueFirst) {
+      sum = 0;
+
+      return sum;
+    }
+
+    if (valueFirst || valueFirst === 0) {
```

Comment:
The condition 'valueFirst || valueFirst === 0' is correct for checking if a number is passed, but the use of 'i' is unnecessary.